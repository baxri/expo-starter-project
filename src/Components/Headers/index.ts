import AuthHeader from './AuthHeader';
import BrandedHeader from './Branded';
import MainHeader from './Main';

export { MainHeader, BrandedHeader, AuthHeader };
