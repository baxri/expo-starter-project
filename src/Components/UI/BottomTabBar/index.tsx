import { View } from 'react-native';
import { CommonActions } from '@react-navigation/native';
import Transfer from 'assets/images/Tabs/transfer.svg';

import map from 'lodash/map';

import { SafeAreaView } from 'Services/safeArea';

import { TabButton, TabContainer, TabTitle, TransferContainer } from './styles';

function BottomTabBar({ state, descriptors, navigation }: any) {
  const handleTabPress = (route: any) => {
    const event = navigation.emit({
      type: 'tabPress',
      target: route.key,
      canPreventDefault: true,
    });

    if (!event.defaultPrevented) {
      navigation.dispatch({
        ...CommonActions.navigate(route.name),
        target: state.key,
      });
    }
  };

  const tabsEl = map(state?.routes, (route, index) => {
    const { options } = descriptors[route.key];
    const isFocused = state.index === index;

    return (
      <TabButton
        activeOpacity={0.9}
        key={index}
        onPress={() => handleTabPress(route)}
      >
        {isFocused && <View />}
        <View>
          {options.tabBarIcon({ color: isFocused ? '#101F48' : '#C5C8D2' })}
        </View>
        <TabTitle color={isFocused ? '#101F48' : '#6D758E'}>
          {options.title}
        </TabTitle>
      </TabButton>
    );
  });

  return (
    <View
      style={{
        backgroundColor: '#FBFCFF',
        position: 'absolute',
        bottom: 0,
        left: 0,
        right: 0,
        borderTopWidth: 0.5,
        borderTopColor: '#EFF5FF',
        borderTopLeftRadius: 24,
        borderTopRightRadius: 24,
      }}
    >
      <TabContainer>
        {tabsEl}
        <TransferContainer
          activeOpacity={0.9}
          onPress={() => {
            navigation.navigate('Deposit');
          }}
        >
          <Transfer />
        </TransferContainer>
      </TabContainer>

      <SafeAreaView bottom />
    </View>
  );
}

export default BottomTabBar;
