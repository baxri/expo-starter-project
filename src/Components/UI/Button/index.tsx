import React from 'react';
// @ts-ignore
import { pick } from '@styled-system/props';

import { Container, ContainerProps, Indicator, Inner, Title } from './styles';

type Props = ContainerProps & {
  disabled?: boolean;
  link?: boolean;
  loading?: boolean;
  secondary?: boolean;
  title: string;
  onPress?: () => void;
};

function Button({
  disabled = false,
  link = false,
  title,
  loading = false,
  secondary = false,
  onPress,
  ...rest
}: Props) {
  return (
    <Container
      {...pick(rest)}
      activeOpacity={0.8}
      disabled={disabled || loading}
      onPress={onPress}
    >
      {({ pressed }: { pressed: boolean }) => (
        <Inner
          disabled={disabled}
          link={link}
          pressed={pressed}
          secondary={secondary}
        >
          {loading ? (
            <Indicator link={link} secondary={secondary} />
          ) : (
            <Title
              disabled={disabled}
              link={link}
              numberOfLines={2}
              pressed={pressed}
              secondary={secondary}
            >
              {title}
            </Title>
          )}
        </Inner>
      )}
    </Container>
  );
}

export default Button;
