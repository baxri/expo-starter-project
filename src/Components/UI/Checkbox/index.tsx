import React from 'react';
// @ts-ignore
import { pick } from '@styled-system/props';

import { Box, Checkmark, Container, ContainerProps, Label } from './styles';

type Props = ContainerProps & {
  disabled?: boolean;
  error?: boolean | string;
  label?: string;
  value: boolean;
  onChange?: (value: boolean) => void;
};

function Checkbox({
  label,
  value,
  error = false,
  disabled = false,
  onChange,
  ...rest
}: Props) {
  const checked = !!value;

  return (
    <Container
      {...pick(rest)}
      disabled={disabled}
      onPress={() => onChange?.(!value)}
    >
      <Box checked={checked} disabled={disabled} error={error}>
        {checked && <Checkmark disabled={disabled} />}
      </Box>

      {!!label && (
        <Label disabled={disabled} error={error} ml={3}>
          {label}
        </Label>
      )}
    </Container>
  );
}

export default Checkbox;
