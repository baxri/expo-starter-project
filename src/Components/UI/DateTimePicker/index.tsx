import React, { useCallback, useMemo, useState } from 'react';
import { DateTime } from 'luxon';

import Label from './Label';
import Sheet from './Sheet';
import {
  AccessoryContainer,
  Container,
  ContainerProps,
  DateAccessory,
  Inner,
  StyledFieldHelper,
  TimeAccessory,
  Value,
  ValueContainer,
} from './styles';

type Props = ContainerProps & {
  disabled?: boolean;
  error?: boolean | string;
  helper?: string;
  label?: string;
  required?: boolean;
  time?: boolean;
  minimumValue?: DateTime;
  value?: DateTime;
  onChange?: (value: DateTime) => void;
};

function DateTimePicker({
  label = '',
  value,
  error = false,
  helper = '',
  disabled = false,
  required = false,
  time = false,
  minimumValue,
  onChange,
  ...rest
}: Props) {
  const [sheetVisible, setSheetVisible] = useState(false);

  const valueStr = useMemo(() => {
    if (!value) {
      return null;
    }

    return time
      ? value.toLocaleString(DateTime.TIME_SIMPLE)
      : value.toLocaleString(DateTime.DATE_SHORT);
  }, [value, time]);

  const handleSheetRequestClose = useCallback(() => {
    setSheetVisible(false);
  }, []);

  return (
    <Container {...rest} onPress={() => setSheetVisible(true)}>
      <Inner error={error}>
        {!!label && (
          <Label
            active={!!valueStr}
            disabled={disabled}
            label={label}
            required={required}
          />
        )}

        <ValueContainer>{valueStr && <Value>{valueStr}</Value>}</ValueContainer>
        <AccessoryContainer>
          {time ? <TimeAccessory /> : <DateAccessory />}
        </AccessoryContainer>
      </Inner>

      {helper || typeof error === 'string' ? (
        <StyledFieldHelper error={error} value={helper} />
      ) : null}

      <Sheet
        minimumValue={minimumValue}
        time={time}
        value={value}
        visible={sheetVisible}
        onChange={onChange}
        onRequestClose={handleSheetRequestClose}
      />
    </Container>
  );
}

export default DateTimePicker;
