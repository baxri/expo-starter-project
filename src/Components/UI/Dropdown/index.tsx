import React, { ComponentProps, useMemo, useState } from 'react';

import find from 'lodash/find';
import isNil from 'lodash/isNil';

import ListSheet from 'Components/UI/ListSheet';

import Label from './Label';
import {
  Chevron,
  ChevronContainer,
  Container,
  ContainerProps,
  Inner,
  StyledFieldHelper,
  Value,
  ValueContainer,
} from './styles';

type ListSheetProps = ComponentProps<typeof ListSheet>;

type Props = ContainerProps & {
  disabled?: boolean;
  error?: boolean | string;
  helper?: string;
  label?: string;
  required?: boolean;
  value?: string | null;
  options?: ListSheetProps['options'];
  onChange?: (value: string) => void;
};

function Dropdown({
  label = '',
  value = '',
  error = false,
  helper = '',
  options,
  disabled = false,
  required = false,
  onChange,
  ...rest
}: Props) {
  const [sheetVisible, setSheetVisible] = useState(false);

  const handleSheetClose = () => {
    setSheetVisible(false);
  };
  const handleSheetSelect = (option?: { id: string; title: string }) => {
    if (option && onChange) {
      onChange(option.id);
    }
  };

  const selectedOption = useMemo(() => {
    if (isNil(value)) {
      return null;
    }

    return find(options, { id: value });
  }, [value, options]);

  return (
    <Container {...rest} onPress={() => setSheetVisible(true)}>
      <Inner error={error}>
        {!!label && (
          <Label
            active={!!selectedOption}
            disabled={disabled}
            label={label}
            required={required}
          />
        )}

        <ValueContainer>
          {selectedOption && (
            <Value numberOfLines={1}>{selectedOption.title}</Value>
          )}
        </ValueContainer>

        <ChevronContainer>
          <Chevron />
        </ChevronContainer>
      </Inner>

      {helper || typeof error === 'string' ? (
        <StyledFieldHelper error={error} value={helper} />
      ) : null}

      <ListSheet
        options={options}
        value={value}
        visible={sheetVisible}
        onClose={handleSheetClose}
        onSelect={handleSheetSelect}
      />
    </Container>
  );
}

export default Dropdown;
