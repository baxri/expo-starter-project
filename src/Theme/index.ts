import DarkTheme from './Dark';
import MainTheme from './Main';

enum ThemeID {
  Main = 'main',
  Dark = 'dark',
}

const THEMES: Record<ThemeID, any> = {
  [ThemeID.Main]: MainTheme,
  [ThemeID.Dark]: DarkTheme,
};

export { ThemeID, THEMES };
