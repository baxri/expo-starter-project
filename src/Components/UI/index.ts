import DefaultTabBar from './TabBar/Default';
import BottomTabBar from './BottomTabBar';
import Button from './Button';
import Checkbox from './Checkbox';
import DateTimePicker from './DateTimePicker';
import Dropdown from './Dropdown';
import FieldHelper from './FieldHelper';
import FieldLabel from './FieldLabel';
import FlatList from './FlatList';
import KeyboardSpacer from './KeyboardSpacer';
import LanguageSelector from './LanguageSelector';
import ListSheet from './ListSheet';
import RadioGroup from './RadioGroup';
import ScrollView from './ScrollView';
import Spinner from './Spinner';
import StatusBar from './StatusBar';
import TabSwitcher from './TabSwitcher';
import Text from './Text';
import TextField from './TextField';
import TouchableOpacity from './TouchableOpacity';
import UserImage from './UserImage';
import { Column, Row } from './View';
import BikeColor from './WelcomeCarusel';

export {
  Text,
  Button,
  DefaultTabBar,
  Column,
  FieldHelper,
  FieldLabel,
  KeyboardSpacer,
  Row,
  ScrollView,
  Spinner,
  TextField,
  StatusBar,
  TouchableOpacity,
  Dropdown,
  ListSheet,
  FlatList,
  Checkbox,
  RadioGroup,
  DateTimePicker,
  BikeColor,
  BottomTabBar,
  LanguageSelector,
  TabSwitcher,
  UserImage,
};
