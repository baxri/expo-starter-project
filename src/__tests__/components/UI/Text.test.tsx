import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components/native';

import { ThemeID, THEMES } from 'Theme';

import { Text } from 'Components/UI';

test('Text renders correctly', () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={THEMES[ThemeID.Main]}>
        <Text>TEXT</Text>
      </ThemeProvider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
