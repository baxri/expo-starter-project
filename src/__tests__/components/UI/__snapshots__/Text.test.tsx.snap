// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text renders correctly 1`] = `
<Text
  color="secondary2"
  fontSize={1}
  fontWeight={1}
  style={
    [
      {
        "color": "#555B6E",
        "fontSize": 12,
        "fontWeight": "500",
      },
    ]
  }
>
  TEXT
</Text>
`;
